api/acme.go

```diff
@@ -0,0 +1,849 @@
+package api
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/api/acme"
+	"github.com/smallstep/cli/jose"
+	"github.com/smallstep/nosql"
+)
+
+func link(url, typ string) string {
+	return fmt.Sprintf("<%s>;\"%s\"", url, typ)
+}
+
+type acmeHandler struct {
+	auth   Authority
+	db     nosql.DB
+	Dir    *acme.Directory
+	domain string
+	prefix string
+}
+
+// newACMEHandler returns a new acmeHandler type.
+func newACMEHandler(auth Authority, db nosql.DB, domain, prefix string, dirOpts *acme.DirectoryOptions) *acmeHandler {
```

Comment:
`api/acme.go` should be "dumb" it doesn't have to have access to the DB, it's function must be:
1. Read data from the request
2. Build the necessary objects
3. Pass those objects to an `Authority` like interface
4. Get the response of that interface
5. Send the the response to the user.